indico_toolkit/auto_populate/populator.py

```diff
@@ -0,0 +1,294 @@
+import time
+from json import loads
+from typing import List, Tuple, Set
+from indico import IndicoClient
+from indico.types import Workflow
+from indico.queries import CreateExport, DownloadExport, GetDataset, GetModelGroup
+from indico_toolkit.errors import ToolkitPopulationError
+from indico_toolkit.pipelines import FileProcessing
+from indico_toolkit.structure.create_structure import Structure
+
+
+class AutoPopulator:
+    def __init__(self, client: IndicoClient):
+        """
+        Module for manipulating and creating new workflows and teach tasks.
+
+        Args:
+            client (IndicoClient): instantiated Indico Client
+        """
+        self.client = client
+        self.structure = Structure(client)
+        self.file_paths = []
+        self.file_to_class = {}
+        self.file_texts = []
+        self._fp = FileProcessing()
+        self._exceptions = []
+
+    def set_file_paths(
```

Comment:
I'd hide this method `_set_file_paths` unless you intend for it to be called directly by user (and if so-- I'd put example usage in the class docstring so people know to call first)